src/common/browser-adapters/browser-adapter-event-manager.ts

```diff
@@ -51,78 +50,90 @@ export class BrowserAdapterEventManager {
     };
 
     private processDeferredEvents(): void {
-        this.deferredEvents = this.deferredEvents.filter(
-            event => this.processEvent(event.eventType, event.eventArgs, true) === false,
-        );
+        const stillDeferred: DeferredEventDetails[] = [];
+        this.deferredEvents.forEach(deferredEvent => {
+            if (this.eventsToApplicationListenersMapping[deferredEvent.eventType]) {
+                try {
+                    this.processEvent(deferredEvent.eventType, deferredEvent.eventArgs);
+                    deferredEvent.deferredResolution.resolveHook(true);
+                } catch (error) {
+                    console.error(error);
+                    deferredEvent.deferredResolution.rejectHook();
+                }
+            } else {
+                stillDeferred.push(deferredEvent);
+            }
+        });
+        this.deferredEvents = stillDeferred;
     }
 
-    private async forwardEventToApplicationListener(
+    private forwardEventToApplicationListener(
         listener: ApplicationListener,
         eventArgs: any[],
     ): Promise<any> {
-        const result = listener(...eventArgs);
+        let result: any;
+        try {
+            result = listener(...eventArgs);
+        } catch (error) {
+            console.error('Error thrown in application listener: ', error);
+            throw error;
```

Comment:
I don't think rethrowing like this is quite right; I think that even if the listener throws an error, it's probably safest for us to still apply the rest of the logic and return a promise to the browser event (otherwise, a single misbehaving event handler is liable to cause the whole service worker to be torn down early).